### PR TITLE
refactor sales page and nav bar

### PR DESCRIPTION
**Pull request by rossneilson**
## Description
I refactored quite a lot here to allow the addition of new features and to clean up the code slightly.
Previously context was used to store state but I moved this to redux to retain the state when navigating the app.
It was added to the page object since it is similar to page metadata.

I also refactored how the components are rendered, reducing complexity with fewer conditional statements and repeating code.
To sort the sales I used a useEffect that triggers on status change and is sorted depending on the sale view. 

## Motivation and Context

#251
#283

## How Has This Been Tested?
Locally and with unit testing for the nav bar.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/views/Sale/components/PurchaseTokensForm/index.tsx [hunk 1]
```diff
@@ -148,6 +148,8 @@ export const PurchaseTokensForm = ({ saleId }: PurchaseTokensFormComponentProps)
         )} ${sale?.tokenOut.symbol}`
       )
     }
+    console.log({ purchaseMaximumAllocation })
+    console.log({ quantity })
```
Comment (mprasanjith) [inline src/views/Sale/components/PurchaseTokensForm/index.tsx:152]: I think we should remove the logs here, if these were not left intentionally?

**Review by mprasanjith — CHANGES_REQUESTED** [review]

File: src/views/Sale/components/PurchaseTokensForm/index.tsx (hunk 1 above)
Comment (rossneilson) [reply src/views/Sale/components/PurchaseTokensForm/index.tsx]: Thanks, I actually removed them but got distracted before committing the changes.
Removed now

Comment (mprasanjith) [reply src/views/Sale/components/PurchaseTokensForm/index.tsx]: Great, everything else looks great.

**Review by mprasanjith — APPROVED** [review]

**Review by martinkrung — APPROVED** [review]
perfect, it feels also faster now!

**Pull request merged**